.github/workflows/test.yml

```diff
@@ -34,14 +34,14 @@ jobs:
           architecture: 'x64'
 
       - run: pip3 install hatch
-      - run: hatch run tests.py3.9-2.7:type-check
+      - run: hatch run tests.py3.9-2.7.3:type-check
 
   Run-Unit-Tests:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         python-version: ["3.8", "3.9", "3.10"]
-        airflow-version: ["2.3", "2.4", "2.5", "2.6", "2.7"]
+        airflow-version: ["2.3.4", "2.4.3", "2.5.3", "2.6.3", "2.7.3", "2.8.1"]
```

Comment:
If we pin micro versions, we need to be aware of every micro release from Airflow and update Cosmos tests. What do you think of sticking to minor releases?

Reply:
Good point, I reverted this and it also solves the issue of using the branch workflow file since it's the same as main.